drivers/source/UnbufferedSerial.cpp

```diff
@@ -99,6 +99,24 @@ short UnbufferedSerial::poll(short events) const
     return revents;
 }
 
+int UnbufferedSerial::enable_input(bool enabled)
+{
+    lock();
```

Comment:
`lock()` is not required, as the base class does it. (It isn't in the existing `set_flow_control` either, but is for the functions calling `_base_putc` etc). 

(I note a minor bug in `SerialBase::set_flow_control` - it's setting `_set_flow_control_dp_func` outside its `lock()`.)

Hadn't noticed the return value difference - interesting. Doesn't matter.

Kind of clarifies a bit to me why the `using` would not be specified to override the virtuals - `using` can be there to choose between mismatching functions in different bases.